packages/ruleset/src/utils/merge-allof-schema-properties.js

```diff
@@ -3,28 +3,52 @@ const { isObject, mergeWith } = require('lodash');
 // Takes a schema, and if an allOf field is provided,
 // merges all allOf schema properties to create one schema
 module.exports = function(schema) {
-  const mergedSchema = Object.assign({}, schema);
-  const allOfArr = mergedSchema['allOf'];
-  // delete now, so it is not merged back into the final result
-  delete mergedSchema['allOf'];
+  // Our merge target is initially an empty object.
```

Comment:
Renovated this utility function so that the logic works the same as the SDK generator in terms of how an allOf is processed.  The main change here is to start with an empty object as the target schema, then merge in each allOf list element, one by one.  Then as a final step, merge in the original schema (with its allOf list removed).  This ensures that each successive allOf list element will overwrite values from previous allOf list elements, and the main schema's own field values will win over any of the allOf list element schemas.